R/prune.R

```diff
@@ -0,0 +1,158 @@
+    }
+    db_entries
+  }
+
+  modelling_group <- thing_exists(e$prune_csv$modelling_group, "modelling_group")
+  disease <- thing_exists(e$prune_csv$disease, "disease")
+  touchstone <- thing_exists(e$prune_csv$scenario, "scenario_description")
+  scenario <- thing_exists(e$prune_csv$touchstone, "touchstone")
+
+
+  # For each row in the CSV file:
+
+  burden_estimate_set <- NULL
+
+  for (i in seq_len(nrow(e$prune_csv))) {
+    row <- e$prune_csv[i, ]
+    touchstone <- unlist(strsplit(row$touchstone, ";"))
+    modelling_group <- unlist(strsplit(row$modelling_group, ";"))
+    disease < unlist(strsplit(row$disease, ";"))
+    scenario <- unlist(strsplit(row$scenario, ";"))
+
+    touchstone <- if ("*" %in% touchstone) NULL else touchstone
+    modelling_group <- if ("*" %in% modelling_group) NULL else modelling_group
+    disease <- if ("*" %in% disease) NULL else disease
+    scenario <- if ("*" %in% scenario) NULL else scenario
+
+    # Find all the responsibilities for any of these fields, including
+    # the current_burden_estimate_set
+
+    resp_info <- DBI::dbGetQuery(con, sprintf("
+      SELECT modelling_group, disease,
+             scenario.touchstone as touchstone,
+             scenario.scenario_description as scenario,
+             responsibility.id as responsibility,
+             current_burden_estimate_set
+        FROM responsibility
+        JOIN responsibility_set
+          ON responsibility_set.id = responsibility.responsibility_set
+        JOIN scenario
+          ON responsibility.scenario = scenario.id
+        JOIN scenario_description
+          ON scenario_description.id = scenario.scenario_description
+          %s", get_filters(touchstone, modelling_group, disease, scenario)))
+
+    # Ignore where no burden_estimate_set has been uploaded.
+
+    resp_info <- resp_info[!is.na(resp_info$current_burden_estimate_set), ]
+
+    #################################################################
+    # Find all the burden estimate sets for the above
+    # responsibilities, that aren't the current_burden_estimate_set
+
+    burden_estimate_set <- rbind(burden_estimate_set,
+      DBI::dbGetQuery(con, sprintf("
+        SELECT *
+          FROM burden_estimate_set
+         WHERE responsibility IN %s
+           AND NOT id IN %s",
+        sql_in(c(-1, resp_info$responsibility)),
+        sql_in(c(-1, resp_info$current_burden_estimate_set)))))
+  }
+
+  # We now have a list of burden estimate sets to cull -
+  # Remove any duplicates, and return
+
+  burden_estimate_set <- burden_estimate_set[!duplicated(burden_estimate_set$id), ]
+  list(burden_estimate_set = burden_estimate_set)
+}
+
+test_extract_prune <- function(e) {
+
+}
+
+transform_prune <- function(extracted_data) {
+  if ("burden_estimate_set" %in% names(extracted_data)) {
+    list(burden_estimate_set = extracted_data$burden_estimate_set)
+  } else {
+    NULL
+  }
+}
+
+test_transform_prune <- function(t) {
+
+}
```

Comment:
Also missing impl?

Reply:
There's nothing really to test here as the transform stage doesn't do anything - it just passes through the extracted data unchanged, (or NULL if it was NULL)

Reply:
Added a comment to explain